### ISSUE TITLE
Generated Go Code Can Import Its Own Package

### ISSUE THREAD

**Issue by FrankJLhota** [micro/protoc-gen-micro#12]
Assume I have a package `myorg/myprog/mypkg` with two protobuf source files: `msg.proto` and `service.proto`. Assume further that `service.proto` has the following directive:
```
import "msg.proto";
```

Then the go code generated for `service.proto` will have the line:
```
import _ "."
```
Needless to say, the go compiler fails to build with this, generating the error message `local import "." in non-local package`.

Could this be fixed by using the full path for the included proto file? If we revise the `service.proto` include directive to read
```
import "myorg/myprog/mypkg/msg.proto";
```
then the go code generated by `protoc_gen_micro` for this file would then have the line
```
import _ "myorg/myprog/mypkg"
```
This creates another problem: this go file is importing its own package, creating an import cycle.

AFAICT `protoc-gen-micro` cannot easily generate correct go code for any proto file that includes another proto file in the same directory, if at all.

**Comment by asim** [comment]
Do not include protos in the same directory. What's the point in this? They're in the same directory. Imagine it as go code. A package would not import itself. It already has reference to its own package. Your protoc command needs to specify the proto path and files to be generated at the same time. Generate the code for a single directory in one command. 

To me this is a non issue. Does this issue present itself if you use gRPC code generation? I imagine the reason that would likely work is because they generate one file whereas we have some separation of code here. 

Basically do not include files in the same directory. Just specify them to be built at the same time on the command line

**Comment by FrankJLhota** [comment]
Sorry, but that does not work. If the needed proto file is specified on the command line but not in an `include` directive, then protoc fails with an error message of the form
```
myorg/myprog/mypkg/service.proto:123:9: "myorg.myprog.mypkg.Address" seems to be
defined in "myorg/myprog/mypkg/msg.proto", which is not imported by
"myorg/myprog/mypkg/service.proto".  To use it here, please add the necessary
import.
```

**Comment by asim** [comment]
Here's a working example. Both files `foo.proto` and `bar.proto` are in the same directory with the same package name `foo`. The `bar.proto` file imports `foo.proto`. The issue you've highlighted exists whether it's micro code generation or the default go or even grpc code generation. This is my advised method.

Command
```
protoc --proto_path=. --micro_out=. --go_out=plugins=grpc:. foo.proto bar.proto 
```

foo.proto
```
syntax = "proto3";

package foo;

service Foo {
	rpc Bar(Request) returns (Response) {}
}

message Request {
	string name = 1;
}

message Response {
	string msg = 1;
}
```

bar.proto
```
syntax = "proto3";

package foo;

import "foo.proto";

service Bar {
	rpc Baz(Request) returns (Response) {}
}
```

**Comment by asim** [comment]
Closing as I believe its a non-issue.
